extensions/amp-viewer-integration/0.1/amp-viewer-integration.js

```diff
@@ -56,31 +57,76 @@ export class AmpViewerIntegration {
     dev().fine(TAG, 'handshake init()');
     const viewer = viewerForDoc(this.win.document);
     this.unconfirmedViewerOrigin_ = viewer.getParam('origin');
-    if (!this.unconfirmedViewerOrigin_) {
-      dev().fine(TAG, 'Viewer origin not specified.');
-      return null;
+    this.unconfirmedViewerOrigin_ = ''; // mocking what would happen with a native app.
```

Comment:
Don't forget to remove :)

Generally, though, it's best to be explicit here. We already have `viewer.getParam('webview') == '1'` reserved for web views. So we could try to use it here as well.